src/features/events/services/events.service.ts

```diff
@@ -76,20 +78,15 @@ export class EventsService
     contract: Contract,
     data: unknown,
   ) {
-    switch (event) {
-      case CrowdfundingEvent.Launch:
-        await this.campaignLaunchService.create(data);
-        break;
-      case CrowdfundingEvent.Pledge:
-        await this.campaignPledgeService.create(data);
-        break;
-      case CrowdfundingEvent.Claim:
-        await this.campaignClaimService.create(data);
-        break;
-      case CrowdfundingEvent.Refund:
-        await this.campaignRefundService.create(data);
-        break;
-    }
+    const servicesByEvent = {
+      [CrowdfundingEvent.Launch]: this.campaignLaunchService,
+      [CrowdfundingEvent.Pledge]: this.campaignPledgeService,
+      [CrowdfundingEvent.Claim]: this.campaignClaimService,
+      [CrowdfundingEvent.Refund]: this.campaignRefundService,
+      [CrowdfundingEvent.Cancel]: this.campaignCancelService,
+    };
+
```

Comment:
NIT: I think it would be nice to check if the event is tracked. (it should be, but just in case).

```suggestion
if (!servicesByEvent[event]) {
  return console.log('No handler for related event');
}
```